variable/include/scipp/variable/transform.h

```diff
@@ -326,8 +421,24 @@ template <bool dry_run> struct in_place {
       return;
 
     auto run = [&](auto indices, const auto &end) {
-      for (; indices != end; indices.increment())
-        call_in_place(op, indices, arg, other...);
+      const auto inner_strides = indices.inner_strides();
+      if (std::all_of(inner_strides.begin(), inner_strides.end(),
+                      [](auto x) { return x == 0; })) {
+        // The special cases don't work when all strides are 0.
+        for (; indices != end; indices.increment())
+          call_in_place(op, indices, arg, other...);
```

Comment:
Can this ever happen? Strides may be zero:
- in a reduction operation (but then the reduced operand has stride != 0)
- in an operation with broadcast (but then the *other* operand has stride != 0)

Reply:
Yes, when all operands are scalar, i.e.
```.py
a = sc.Variable(value=2)
b = sc.Variable(value=1)
a *= b
```